File: January/Day19/leetcode_974.cpp
class Solution {
public:
    int subarraysDivByK(vector<int>& nums, int k) {
      int count = 0;
      unordered_map<int,int>mp;
      int sum = 0;
      mp[0]++;
      for(auto x:nums)
      {
        sum+=x;
        if(mp[(sum%k+k)%k]>0)
        {
        count+=mp[(sum%k+k)%k];
        }
        mp[(sum%k+k)%k]++;
      }
      return count;  
    }
};

File: January/Day19/leetcode_974.java
class Solution {
    public int subarraysDivByK(int[] nums, int k) {
       int count=0;
       int runningsum=0;
       HashMap<Integer,Integer> map = new HashMap<>();
       map.put(0,1);
       for(int num:nums){
           runningsum+=num;
           int rem=runningsum%k;
           rem=rem<0?rem+k:rem;
           if(map.containsKey(rem)){
               count=count+map.get(rem);
               map.put(rem,map.getOrDefault(rem,0)+1);
           }
           else{
               map.put(rem,1);
           }
       } 
       return count;
    }
}
